src/providers/AuthProvider.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+
+import { AccountStore } from '../stores/AccountStore';
+
+type Props = {
+  children: JSX.Element;
+};
+
+function AuthProvider({ children }: Props): JSX.Element | null {
```

Comment:
I think this is a small enough amount of code that it could just be directly in Root.tsx, but this could also eventually expand to have some other user or account related side effects.  If you are going to keep it as a provider, it should probably follow the Context.Provider structure like the other providers.

Reply:
I changed it back to `useEffect` hook and left the comment. When I understand all the possibles corner cased I will switch to the hook if possible.